modules/nf-core/sentieon/tnhaplotyper2/main.nf

```diff
@@ -0,0 +1,111 @@
+process SENTIEON_TNHAPLOTYPER2 {
+    tag "$meta.id"
+    label 'process_high'
+    label 'sentieon'
+
+    secret 'SENTIEON_LICENSE_BASE64'
+
+    container 'nf-core/sentieon:202112.06'
+
+    input:
+    tuple val(meta), path(input), path(input_index), path(intervals)
+    tuple val(meta2), path(dict)
+    tuple val(meta3), path(fasta)
+    tuple val(meta4), path(fai)
+    tuple val(meta5), path(germline_resource)
+    tuple val(meta6), path(germline_resource_tbi)
+    tuple val(meta7), path(panel_of_normals)
+    tuple val(meta8), path(panel_of_normals_tbi)
+    val(emit_orientation_data)
+    val(emit_contamination_data)
+
+    output:
+    tuple val(meta), path("*.orientation_data.tsv")  , optional:true , emit: orientation_data
+    tuple val(meta), path("*.contamination_data.tsv"), optional:true , emit: contamination_data
+    tuple val(meta), path("*.segments")              , optional:true , emit: contamination_segments
+    tuple val(meta), path("*.stats")                 , emit: stats
+    tuple val(meta), path("*.vcf.gz")                , emit: vcf
+    tuple val(meta), path("*.vcf.gz.tbi")            , emit: index
+    path "versions.yml"                              , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    // Exit if running this module with -profile conda / -profile mamba
+    if (workflow.profile.tokenize(',').intersect(['conda', 'mamba']).size() >= 1) {
+        error "Sentieon modules do not support Conda. Please use Docker / Singularity / Podman instead."
+    }
+    def sentieon_auth_mech_base64 = task.ext.sentieon_auth_mech_base64 ?: ''
+    def sentieon_auth_data_base64 = task.ext.sentieon_auth_data_base64 ?: ''
+    def args                      = task.ext.args                      ?: ''
+    def args2                     = task.ext.args2                     ?: ''
+    def prefix                    = task.ext.prefix                    ?: "${meta.id}"
+    def tumour_id                 = task.ext.tumour_id                 ?: "${meta.tumour_id}"
+    def normal_id                 = task.ext.normal_id                 ?: "${meta.normal_id}"
```

Comment:
Instead of introducing new task.ext args, can we do soemthing like this
```suggestion
    def tumour_id                 = task.ext.args.contains("--tumor_sample")  ?: "${meta.tumour_id}"
    def normal_id                 = task.ext.args.contains("--normal_sample") ?: "${meta.normal_id}"
```

Reply:
hmm ... but isn't `task.ext.args.contains("--tumor_sample")` just a boolean?

Reply:
ah right.. how about this?  
```suggestion
    def tumour_id                 = task.ext.args.contains("--tumor_sample")  ? "" : "${meta.tumour_id}"
    def normal_id                 = task.ext.args.contains("--normal_sample") ? "" : "${meta.normal_id}"
```

Reply:
Do we always need a tumor normal pair, or does a tumor only sample works here too?

Reply:
From the docs - "The TNhaplotyper2 algorithm performs the somatic variant calling on the tumor-normal matched pair or the tumor and panel of normal data, using a Haplotype based algorithm."

https://support.sentieon.com/manual/usages/general/?highlight=tnseq#tnhaplotyper2-algorithm

That seems to indicate that the algorithm does not require a normal-sample if supplied with tumor-sample and a panel-of-normal data.